config/routes.rb

```diff
@@ -233,6 +233,7 @@
         get :manage
       end
     end
+    resources :kudos, only: [:index]
```

Comment:
I would use the better method here and consider changing all the things as a different issue.